src/devices/Mcp3428/Mcp342x.cs

```
@@ -0,0 +1,387 @@
+
+        public int ChannelCount { get; }
+
+        protected ConversionResult LastConversion
+        {
+            get
+            {
+                return _lastConversion;
+            }
+            set
+            {
+                _lastConversion = value;
+                OnConversion?.Invoke(this, _lastConversion);
+            }
+        }
+
+        public event EventHandler<ConversionResult> OnConversion;
+
+        protected void WriteConfig(byte configByte)
+        {
+            _i2cDevice.WriteByte(configByte);
+        }
+
+        private static int _asyncThreshold = 20;
+
+        /// <summary>
+        /// Sets the lower time limit in ms. If the current configuration reads data faster the synchronous API is used.
+        /// This can save some overhead.
+        /// </summary>
+        /// <remarks>
+        /// The default configuration is 20ms. This means that only in 16 bit resolution read waits asynchronously.
+        /// Setting it to 0 or lower disables the function.
+        /// </remarks>
+        /// <param name="thresh">Time limit in ms. Default: 20ms</param>
+        /// <autogeneratedoc />
+        public static void SetAsyncThreshold(int thresh) { _asyncThreshold = thresh; }
+
+        /// <summary>
+        /// One-shot read as an asynchronous operation. Initiates read and waits for it to finish.
+        /// Async API was required as reading in 16bit resolution can take more than 60ms.
+        /// </summary>
+        /// <param name="channel">The channel.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>ValueTask.</returns>
+        /// <exception cref="System.IO.IOException">
+        /// Device is not in One-Shot mode
+        /// or
+        /// ADC Conversion was not ready after {tries}
+        /// </exception>
+        /// <autogeneratedoc />
+        protected ValueTask OneShotReadAsync(int channel = -1, CancellationToken cancellationToken = default)
```

should methods taking `channel` throw when invalid value is used?

---

I think it does in the bit setter function.